-1

Tested at: cd0b67622458ab94e5daf5e03a66fe1572856062

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18364/20182/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalRecHitsDQMClient.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalNoiseRatesClient.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalNoiseRates.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc: In member function 'virtual void HcalRecHitsAnalyzer::analyze(const edm::Event&, const edm::EventSetup&)':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc:581:50: error: 'begin' was not declared in this scope
       for ( const auto & recHit : rhitEB.product() ) {
                                                  ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc:581:50: note: suggested alternatives:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02474/slc6_amd64_gcc530/external/boost/1.63.0-mlhled2/include/boost/serialization/detail/get_data.hpp:23:0,
                 from /cvmfs/cms-ib.cern.ch/nweek-02474/slc6_amd64_gcc530/external/boost/1.63.0-mlhled2/include/boost/serialization/vector.hpp:35,
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalRecHitsDQMClient.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalNoiseRatesClient.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalNoiseRates.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/CaloTowersDQMClient.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc:581:35: error: invalid range expression of type 'const edm::SortedCollection<EcalRecHit, edm::StrictWeakOrdering<EcalRecHit> > *'; did you mean to dereference it with '*'?
      for ( const auto & recHit : rhitEB.product() ) {
                                  ^
                                  *
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_2_X_2017-05-30-1100/src/DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc:593:35: error: invalid range expression of type 'const edm::SortedCollection<EcalRecHit, edm::StrictWeakOrdering<EcalRecHit> > *'; did you mean to dereference it with '*'?
      for ( const auto & recHit : rhitEE.product() ) {
</pre>

--- a/DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc
+++ b/DQMOffline/Hcal/src/HcalRecHitsAnalyzer.cc
@@ -578,9 +578,9 @@ void HcalRecHitsAnalyzer::analyze(edm::Event const& ev, edm::EventSetup const& c
     Handle<EBRecHitCollection> rhitEB;
     if ( ev.getByToken(tok_EB_, rhitEB) ) {
       
-      for ( const auto & recHit : rhitEB.product() ) {
+      for ( const auto & recHit : *(rhitEB.product()) ) {
 	
-	double en  = recHit->energy();
+	double en  = recHit.energy();
 	eEcal  += en;
 	eEcalB += en;
       }      
@@ -590,9 +590,9 @@ void HcalRecHitsAnalyzer::analyze(edm::Event const& ev, edm::EventSetup const& c
     Handle<EERecHitCollection> rhitEE;
     if ( ev.getByToken(tok_EE_, rhitEE) ) {
     
-      for ( const auto & recHit : rhitEE.product() ) {
+      for ( const auto & recHit : *(rhitEE.product()) ) {
 	
-	double en   = recHit->energy();
+	double en   = recHit.energy();
 	eEcal  += en;
 	eEcalE += en;
 	

